topdown/json.go

```diff
@@ -130,13 +131,10 @@ func builtinJSONFilter(_ BuiltinContext, operands []*ast.Term, iter func(*ast.Te
 	}
 
 	// Actually do the filtering
-	filterObj := pathsToObject(filters)
-	r, err := obj.Filter(filterObj)
-	if err != nil {
-		return err
-	}
+	et := edittree.NewEditTree(ast.NewTerm(obj))
+	r := et.Filter(filters)
```

Comment:
Here we are just ensuring the existing tests pass and there are no new cases to add.

Reply:
Correct, no new tests should be needed. :slightly_smiling_face: 

I checked the `(*object).Filter` implementation to see where it could return errors, and the main error cases looked to be due to implementation details of the Object-based filtering approach that was being used.

Since the `EditTree`-based `Filter` call doesn't have the same error cases, I made the call always return a `Term` or `nil`. As a result there's no additional error checking or tests needed there, I think.